python/ray/serve/_private/logging_utils.py

```
@@ -121,6 +122,106 @@ def filter(self, record):
+        self._shutdown = False
+        self.flush_timeout_s = flush_timeout_s
+        self._check_thread = None
+        self._check_event = threading.Event()  # Event to notify check thread
+
+        if self.flush_timeout_s is not None and self.flush_timeout_s > 0:
+            self._start_check_thread()
+
+    def _start_check_thread(self):
+        """Start the background thread that checks for idle timeout."""
+        if self._shutdown:
+            return
+
+        # Stop existing thread if any
+        if self._check_thread is not None and self._check_thread.is_alive():
+            return
+
+        # Start background check thread
+        self._check_thread = threading.Thread(
+            target=self._check_idle_timeout,
+            daemon=True,
+            name="TimedMemoryHandler-check",
+        )
+        self._check_thread.start()
+
+    def _check_idle_timeout(self):
+        """Check if idle timeout has elapsed and flush if needed."""
+        while not self._shutdown:
+            try:
+                # Wait for notification or timeout
+                # If no emit happens within flush_timeout_s, wait will timeout
+                event_set = self._check_event.wait(timeout=self.flush_timeout_s)
+
+                # Check shutdown flag
+                if self._shutdown:
+                    break
+
+                # If event was set, it means a new emit happened (timeout was reset)
+                if event_set:
+                    # Clear the event for next wait
+                    self._check_event.clear()
+                    continue
+
+                # If we reach here, wait() timed out (no new logs during the wait period)
+                self.flush()
+
+            except Exception:
+                # Ignore exceptions in background thread to prevent crashes
+                # but check shutdown flag to exit gracefully
+                if self._shutdown:
+                    break
```

We should at least log exception here instead of silently swallowing it. How is this handled in core python logging library?